src/warnings.cpp

```diff
@@ -12,19 +12,22 @@
 
 CCriticalSection cs_warnings;
 std::string strMiscWarning;
+int64_t timestampWarning;
 bool fLargeWorkForkFound = false;
 bool fLargeWorkInvalidChainFound = false;
 
-void SetMiscWarning(const std::string& strWarning)
+void SetMiscWarning(const std::string& strWarning, int64_t timestamp)
 {
     LOCK(cs_warnings);
     strMiscWarning = strWarning;
+    timestampWarning = timestamp;
 }
 
-std::string GetMiscWarning()
+std::pair<std::string, int64_t> GetMiscWarning()
 {
     LOCK(cs_warnings);
-    return strMiscWarning;
+    std::pair<std::string, int64_t> misc(strMiscWarning, timestampWarning);
+    return misc;
```

Comment:
I think this can just be:
```suggestion
    return std::make_pair(strMiscWarning, timestampWarning);
```